### PR TITLE
sys/sysctl.h no longer supported, won't build.

### PR DESCRIPTION
**Pull request by Shadorain**
**Detailed description**

This PR fixes problems with the `fs-squash` plugin. The plugin fails to build because of a missing library: `sys/sysctl.h` which is
deprecated and removed in glibc 2.32. The new and fixed version is `linux/sysctl.h` which I replaced in the plugin. Replacing this led to other build errors so I had to resolve those as well. 
I still have a pre-processor set up that checks the user's glibc build version in case they are using an older glibc version (< `glibc_2.32`). Should build fine with the deprecated `sys/sysctl.h` for those users.

...

**Test plan**

The only test I can think of would be to try building/installing `fs-squash` with `r2pm` before my PR, and then test with my PR and see that it actually will build and work fine.

...

### CODE REVIEW

File: libr/fs/squashfs/unsquashfs.c [hunk 1]
```diff
@@ -1898,14 +1905,14 @@ void initialise_threads(int fragment_buffer_size, int data_buffer_size) {
 			"\n");
 
 	if (processors == -1) {
-#if __linux__
+#if __GLIBC__ == 2 && __GLIBC_MINOR__ <= 30
 		processors = sysconf(_SC_NPROCESSORS_ONLN);
```
Comment (trufae) [inline libr/fs/squashfs/unsquashfs.c:1909]: I think its safe to remove this code. R2 doesnt needs to get the number of processors of anything. Its just a filesystem. Can u remove it? Nice catch!

Comment (Shadorain) [reply libr/fs/squashfs/unsquashfs.c:1909]: For sure!

**Comment by Shadorain** [comment]
That merge commit should not be there....

**Comment by trufae** [comment]
Rhe variable processors is -1 which resulrs on an invalid malloc after the removed code. Remove the global and always use 1

**Comment by Shadorain** [comment]
> Rhe variable processors is -1 which resulrs on an invalid malloc after the removed code. Remove the global and always use 1

gotcha good catch, I can remove the global or I can just set it to `1` and remove anywhere that changes `processors`? how bout the `-processors` flag?

**Comment by trufae** [comment]
Removing all the use cases of it and assume 1 is fine. Sorry i missed your comment

**Comment by Shadorain** [comment]
Oh goodness, I'm sorry, I have no idea how to squash all those commits together (didn't know if I was supposed to as well).

**Comment by trufae** [comment]
Easiest way is to get the hashes of the commits you did into the clipboard and then git reset the branch and pull from master, after this use `git cherry-pick ...hash..` and push -f, if you are worried about losing the changes you can do another PR using a separate branch on a clean checkout

**Pull request merged**